### ISSUE TITLE
Feature Request: Improved Semantic Outline

### ISSUE THREAD

**Issue by MartinHaeusler** [TypeStrong/atom-typescript#1021]
The semantic outline for TypeScript is quite good already. However, it would be even more useful if it would:
- ... display "export" members in a visually different style (icon, color...)
- ... use a consistent set of icons (see Eclipse, VisualStudio...)
- ... support tree-style collapsing of "inner" members (these are indented already)
- ... show the type of variables
- ... show the parameter and return types of functions
- ... be expandable (currently has fixed width)

Basically I would like to see it working like the Eclipse outline for Java:
![outline](https://cloud.githubusercontent.com/assets/16356998/16942947/4c382adc-4d99-11e6-8703-a8a64a1ca62d.png)
